Services/Devices.cs

```diff
@@ -357,19 +368,124 @@ public async Task DeleteListAsync(IEnumerable<string> deviceIds)
+        public async Task<string> CreateListUsingJobsAsync(IEnumerable<string> deviceIds)
+        {
+            this.instance.InitRequired();
+
+            this.log.Info("Starting bulk device creation");
+
+            // List of devices
+            var serializedDevices = new List<string>();
+            foreach (var deviceId in deviceIds)
+            {
+                var device = new ExportImportDevice
+                {
+                    Id = deviceId,
+                    ImportMode = ImportMode.CreateOrUpdate,
+                    Authentication = new AuthenticationMechanism
+                    {
+                        Type = AuthenticationType.Sas,
+                        SymmetricKey = new SymmetricKey
+                        {
+                            PrimaryKey = this.fixedDeviceKey,
+                            SecondaryKey = this.fixedDeviceKey
+                        }
+                    },
+                    Status = DeviceStatus.Enabled,
+                    Tags = new TwinCollection { [SIMULATED_TAG_KEY] = SIMULATED_TAG_VALUE }
+                };
+
+                serializedDevices.Add(JsonConvert.SerializeObject(device));
+            }
+
+            CloudBlockBlob blob;
+            try
+            {
+                blob = await this.WriteDevicesToBlobAsync(serializedDevices);
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Failed to create blob file required for the device import job", e);
+                throw new ExternalDependencyException("Failed to create blob file", e);
+            }
+
+            // Create import job
+            JobProperties job;
+            try
+            {
+                var sasToken = this.GetSasTokenForImportExport();
+                this.log.Info("Creating job to import devices");
+                job = await this.registry.ImportDevicesAsync(blob.Container.StorageUri.PrimaryUri.AbsoluteUri + sasToken, blob.Name);
+                this.log.Info("Job to import devices created");
+            }
+            catch (JobQuotaExceededException e)
+            {
+                this.log.Error("Job quota exceeded, retry later", e);
+                throw new ExternalDependencyException("Job quota exceeded, retry later", e);
+            }
+            catch (Exception e)
+            {
+                this.log.Error("Failed to create device import job", e);
+                throw new ExternalDependencyException("Failed to create device import job", e);
+            }
+
+            return job.JobId;
+        }
+
+        // Check if an IoT Hub job is complete
+        public async Task<bool> IsJobCompleteAsync(string jobId, Action recreateJobSignal)
+        {
+            this.instance.InitRequired();
+
+            JobProperties job;
+
+            try
+            {
+                job = await this.registry.GetJobAsync(jobId);
+
+                switch (job.Status)
```

Comment:
Do we need to care about the default case for this switch?

Reply:
not needed, all expected cases are handled here, and the method throws an exception if something unexpected is returned